### DISCUSSION TITLE
Set Variable inside Event

### DISCUSSION THREAD

**Discussion by codevApp** [alpinejs/alpine#4056] [1. Help]
Hi, 
in the following component I try to set the selected variable in the slide.on event. the log shows correctly but the variable in the Alpine does not react. Outside of the splide.on it works. can anyone help?  

         <div
                x-data="{
                    selected: 0,
                    init() {

                            var splide = new Splide(this.$refs.splide, {
                                perPage: 1,
                                gap: '0.5rem',
                                pagination: true,
                            });
                            splide.mount()
                            this.$watch('selected', () => splide.go(this.selected))

                            splide.on('move', function () {
                               this.selected = splide.index;
                               console.log(this.selected);
                            });

                    },
               }">

**Comment by ekwoka** [comment]
Just remove the `function` keyword and make it an arrow function.

Like you did for the `$watch`.

Arrow functions lack a lexically bound `this`, while the `function` keyword does bind a `this`. This means thats the `this` inside the `function` is not the `this` outside the `function`.

Which `this` is this `this`? Who knows.

(methods directly on objects like you did with `init(){}` are also lexically bound)

**Reply by codevApp** [reply]
Wow, thank you! It´s working.
